### PR TITLE
BUG: DataFrame(ndarray, dtype=categoricaldtype)

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/tests/frame/test_constructors.py [hunk 1]
```diff
@@ -2102,18 +2114,22 @@ def test_constructor_categorical(self):
...
 
+    def test_from_nested_listlike_mixed_types(self):
         # mixed
         df = DataFrame([Categorical(list("abc")), list("def")])
         expected = DataFrame(
             {0: Series(list("abc"), dtype="category"), 1: list("def")}, columns=[0, 1]
         )
         tm.assert_frame_equal(df, expected)
 
+    def test_construct_from_listlikes_mismatched_lengths(self):
```
Comment (jreback) [inline pandas/tests/frame/test_constructors.py:2125]: may pay to split out this test file as getting kind of big

**Review by jreback — APPROVED** [review]
lgtm. let's just re-run ci again

File: pandas/tests/frame/test_constructors.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/frame/test_constructors.py:2125]: ive been thinking that myself.  will address before long

**Comment by jbrockmendel** [comment]
this broke some things locally, am troubleshooting now

**Review by jreback — CHANGES_REQUESTED** [review]
lgtm. can you merge master as i think the tests are now moved

**Comment by jbrockmendel** [comment]
rebased + green

**Review by jreback — APPROVED** [review]

**Pull request merged**